arangod/RocksDBEngine/ReplicatedRocksDBTransactionState.cpp

```diff
@@ -71,6 +71,8 @@ futures::Future<Result> ReplicatedRocksDBTransactionState::doCommit() {
 
   if (!mustBeReplicated()) {
     Result res;
+    // We are aborting the transaction, the list of collections is complete - no
+    // need to acquire the lock.
     for (auto& col : _collections) {
```

Comment:
I would actually be in favor to acquire a read lock here, very close around the _collections usage, not for the full method.

With the APIs we use here we rely on the outside world to "know" when calling a method here is save.
E.g. we rely on the callers to protect es against a concurrent: addCollection() and commit() call.
For this example it may be ok. But for abort in my opinion it is not clear enough. (e.g. I could send many parallel requests, where the first throws an error triggering an abort to other participants, which than races with the initial search request)

I would suggest we better be save than sorry here.